src/app/pages/quizzes-catalog/quizzes-catalog.component.ts

```diff
@@ -16,21 +16,35 @@ import { StoreService } from '../../services/store.service';
   templateUrl: './quizzes-catalog.component.html',
   imports: [ UIKitModule, CommonModule ],
 })
-export class QuizzesCatalogComponent implements OnInit {
+export class QuizzesCatalogComponent implements OnInit, OnDestroy {
   categories$!: Observable<CategoryModel[]>;
   isLoading$!: Observable<boolean>;
+  subscription!: Subscription;
   
   constructor(private storeService: StoreService) { }
-
+  
   ngOnInit(): void {
+    this.subscription = this.storeService.getCategories()
+    .pipe(take(1))
+    .subscribe((categories) => {
+      if (!categories || categories.length === 0) {
+        this.storeService.loadCategories();
+      }
+    });
+    
     this.categories$ = this.storeService.getCategories();
     this.isLoading$ = this.storeService.getLoading();
-    this.storeService.loadCategories();
   }
-
+  
   getCardStyle(index: number): CardItemStyle {
     const styleIndex = index + Math.floor(index / 5);
     const enumValues = Object.values(CardColors);
     return cardItemStyles[enumValues[styleIndex % enumValues.length]];
   }
+
+  ngOnDestroy(): void {
+    if (this.subscription) {
```

Comment:
If you use subscriptions you can try to use an angular operator called takeUntilDestroyed instead of directly unsubscribing in OnDestroy